### PR TITLE
emulator for Phase-2 GCT calo jets and taus

### PR DESCRIPTION
**Pull request by pallabidas**
#### PR description:

reviewed in https://github.com/cms-l1t-offline/cmssw/pull/1161 and https://github.com/cms-l1t-offline/cmssw/pull/1174

- adds a data format: l1tp2::Phase2L1CaloJetCollection
- depends on PR https://github.com/cms-l1t-offline/cmssw/pull/1069 for input GCT towers collection (ECAL clusters + HCAL towers)
- refer to presentation here: https://indico.cern.ch/event/1288569/timetable/?view=standard#68-calorimeter-trigger-firmwar
- not a backport

### CODE REVIEW

**Comment by epalencia** [comment]
Please test

**Comment by epalencia** [comment]
Please test

**Comment by rovere** [comment]
@SohamBhattacharya FYI

**Comment by aloeliger** [comment]
+l1
- Already reviewed in L1 fork.

**Comment by srimanob** [comment]
@cmsbuild please test

Retest after 2 weeks.

**Comment by aloeliger** [comment]
@cms-sw/upgrade-l2 Gentle ping on this. L1 developers would like to use this.

File: L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloJetEmulator.cc [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+    edm::LogError("Phase2L1CaloJetEmulator") << "Failed to get HcalTrigPrimDigi for HF!";
+  iEvent.getByToken(hfToken_, hfHandle);
+  float hfTowers[nHfEta][nHfPhi];
+  float hfEta[nHfEta][nHfPhi];
+  float hfPhi[nHfEta][nHfPhi];
+
+  for (int iphi = 0; iphi < nHfPhi; iphi++) {
+    for (int ieta = 0; ieta < nHfEta; ieta++) {
+      hfTowers[ieta][iphi] = 0;
+      int temp = ieta;
```
Comment (srimanob) [inline L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloJetEmulator.cc:166]: Small thing on static check, you may fix it now.

```
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_0_X_2023-11-21-2300/src/L1Trigger/L1CaloTrigger/plugins/Phase2L1CaloJetEmulator.cc:166:11: warning: Value stored to 'temp' during its initialization is never read [deadcode.DeadStores]
  166 |       int temp = ieta;
```

**Comment by aloeliger** [comment]
please test

**Comment by aloeliger** [comment]
+l1

- Already reviewed in L1 fork.

File: L1Trigger/L1CaloTrigger/interface/Phase2L1CaloJetEmulator.h [hunk 2]
```diff
@@ -0,0 +1,495 @@
+#ifndef _PHASE_2_L1_CALO_JET_EMULATOR_H_
```
Comment (srimanob) [inline L1Trigger/L1CaloTrigger/interface/Phase2L1CaloJetEmulator.h:1]: This header protection does not follow the style in cms coding:
https://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1

**Comment by srimanob** [comment]
+Upgrade

**Comment by epalencia** [comment]
+l1

**Comment by rappoccio** [comment]
+1

**Pull request merged**